cmd/destroy.go

```diff
@@ -111,6 +205,134 @@ func destroyTerraformGroup(groupDir string) error {
+	// to get the exact network names.
+	filter := fmt.Sprintf("name eq \".*%s.*\"", deploymentName)
+	logging.Info("Using wildcard network filter: %s", filter)
+	networks, err := computeService.Networks.List(projectID).Filter(filter).Do()
+	if err != nil {
+		return fmt.Errorf("failed to list networks with wildcard filter: %v", err)
+	}
+
+	if len(networks.Items) == 0 {
+		logging.Info("No matching networks found for project %s.", projectID)
+		return nil
+	}
+
+	firewallsToDelete, err := listAssociatedFirewallRules(projectID, computeService, networks.Items)
+	if err != nil {
+		return err
+	}
+
+	if len(firewallsToDelete) == 0 {
+		logging.Info("No firewall rules found to delete for the identified networks.")
+		return nil
+	}
+
+	return confirmAndDeleteFirewallRules(projectID, deploymentName, &computeServiceWrapper{computeService}, firewallsToDelete)
+}
+
+type computeServiceWrapper struct {
+	*compute.Service
+}
+
+func (w *computeServiceWrapper) FirewallsDelete(projectID string, firewall string) (*compute.Operation, error) {
+	return w.Firewalls.Delete(projectID, firewall).Do()
+}
+
+// listAssociatedFirewallRules lists firewall rules associated with a given set of networks.
+func listAssociatedFirewallRules(projectID string, computeService *compute.Service, networks []*compute.Network) ([]*compute.Firewall, error) {
+	var firewallsToDelete []*compute.Firewall
+	for _, network := range networks {
+		fwList, err := computeService.Firewalls.List(projectID).Filter(fmt.Sprintf("network=\"%s\"", network.SelfLink)).Do()
+		if err != nil {
+			return nil, fmt.Errorf("failed to list firewall rules for network %s: %v", network.Name, err)
+		}
+		firewallsToDelete = append(firewallsToDelete, fwList.Items...)
+	}
+	return firewallsToDelete, nil
+}
+
+// confirmAndDeleteFirewallRules confirms with the user and then deletes the specified firewall rules.
+func confirmAndDeleteFirewallRules(projectID string, deploymentName string, computeService firewallDeleter, firewallsToDelete []*compute.Firewall) error {
+	var firewallNames []string
+	for _, fw := range firewallsToDelete {
+		firewallNames = append(firewallNames, fw.Name)
+	}
+	logging.Info("Found firewall rules to delete: %v", firewallNames)
+
+	if !flagAutoApprove {
+		prompt := fmt.Sprintf("Do you want to delete these %d firewall rules associated with deployment %s? [y/n]: ", len(firewallNames), deploymentName)
+		if !confirmAction(prompt) {
+			logging.Info("Skipping firewall rule deletion.")
+			return nil
+		}
+	}
+
+	logging.Info("Successfully submitted deletion requests for firewall rules.")
+	// Delete firewall rules
+	var deletionErrors []string
+	for _, fwName := range firewallNames {
+		logging.Info("Deleting firewall rule %s...", fwName)
+		_, err := computeService.FirewallsDelete(projectID, fwName)
+		if err != nil {
+			// Log non-critical errors and continue trying to delete other rules
+			msg := fmt.Sprintf("Failed to delete firewall rule %s: %v", fwName, err)
```

Comment:
This message is redundant given the error message  in the next line reads the same. Can directly add the fwName and err to the next log statement